src/v3/src/components/Widget/index.tsx

```diff
@@ -438,23 +439,33 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
     if (isClientTransaction) {
       return;
     }
-    if (widgetRendered && typeof idxTransaction !== 'undefined'
-      && uischema.elements.length > 0) {
-      eventEmitter?.emit?.('afterRender', getEventContext(idxTransaction));
+    if (widgetRendered) {
+      (async () => {
+        const isFirstRender = typeof widgetRenderedOnce.current === 'undefined';
```

Comment:
We can set `widgetRenderedOnce` to be `false` by default when creating and avoid needing do this check:
```ts
const widgetRenderedOnce = useRef<boolean>(false);
```
Then we can directly check the value in the conditional below with `if (widgetRenderedOnce.current)`